backend/src/main/java/com/huellapositiva/application/controller/VolunteerApiController.java

```diff
@@ -137,4 +149,118 @@ public void uploadCurriculumVitae(@RequestPart("cv") MultipartFile cv,
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "Ok, return full information user profile"
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "Bad request, credentials are not valid",
+                            content = @Content()
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @GetMapping("/fetchProfileInformation")
+    @RolesAllowed("VOLUNTEER")
+    @ResponseStatus(HttpStatus.OK)
+    public ProfileDto fetchProfileInformation(@AuthenticationPrincipal String volunteerEmail) {
+        return fetchVolunteerProfileAction.execute(volunteerEmail);
+    }
+
+    @Operation(
+            summary = "Update user profile information",
+            description = "Update user profile information",
+            tags = "user"
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "Ok, return full information user profile"
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "Bad request, credentials are not valid or some field is mandatory",
+                            content = @Content()
+                    ),
+                    @ApiResponse(
+                            responseCode = "409",
+                            description = "Conflict, the new email already match with other email in db",
+                            content = @Content()
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @PostMapping(path = "/updateProfileInformation", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    @RolesAllowed("VOLUNTEER")
+    @ResponseBody
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void updateProfileInformation(@RequestPart("dto") MultipartFile dtoProfileInformation,
+                                        @AuthenticationPrincipal String contactPersonEmail
+                                         ) throws IOException {
+        ProfileDto profileDto = objectMapper.readValue(dtoProfileInformation.getBytes(), ProfileDto.class);
+        try {
+            updateVolunteerProfileAction.execute(profileDto,contactPersonEmail);
+        } catch (IOException ex){
+            throw new SomeFieldIsNullException(ex.getMessage());
```

Comment:
Not necessary, validate in DTO

Reply:
It's done. Please, would you to check it? Thanks.